src/gluonts/dataset/common.py

```diff
@@ -215,11 +215,12 @@ def __init__(
 
     def __iter__(self) -> Iterator[DataEntry]:
         for path in self.files():
-            for line in jsonl.JsonLinesFile(path):
+            for i, line in enumerate(jsonl.JsonLinesFile(path)):
                 data = self.process(line.content)
                 data["source"] = SourceContext(
-                    source=line.span, row=line.span.line
+                    source=line.span.path, row=line.span.line
```

Comment:
I think I accidentally leaked this from latest master branch, however, should be no problem when merging with master.